### PR TITLE
[SDK] Updated `claimConditions.getActive` method to return `null` 

### PR DESCRIPTION
**Pull request by alexmylonas**
### Description

This PR updates the `claimConditions.getActive` method to return `null` when:

- Phase hasn't started yet
- All NFTs for the phase have been claimed 

As described https://github.com/thirdweb-dev/js/issues/237

### CODE REVIEW

**Comment by alexmylonas** [comment]
I can see that the issue https://github.com/thirdweb-dev/js/issues/237 has been closed, but let me know if you still want this, otherwise, I 'll close it!

File: packages/sdk/test/evm/signature-drop-v4.test.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ describe("Signature drop tests (v4)", async () => {
   beforeEach(async () => {
     sdk.updateSignerOrProvider(adminWallet);
 
-    signatureDropContract = await sdk.getSignatureDrop(
+    signatureDropContract = await sdk.getContract(
```
Comment (alexmylonas) [inline packages/sdk/test/evm/signature-drop-v4.test.ts:42]: minor update due to deprecation, this seems to be the last place the [getSignatureDrop](https://github.com/alexmylonas/third-web-js/blob/main/packages/sdk/src/evm/core/sdk.ts#L216) is used internally.

File: packages/sdk/test/evm/signature-drop-v5.test.ts [hunk 2]
```diff
@@ -493,8 +493,8 @@ describe("Signature drop tests (v5)", async () => {
       const conditions = await signatureDropContract.claimConditions.getActive({
         withAllowList: true,
       });
-      invariant(conditions.snapshot);
-      expect(conditions.snapshot[0].address).to.eq(samWallet.address);
+      invariant(conditions?.snapshot);
```
Comment (alexmylonas) [inline packages/sdk/test/evm/signature-drop-v5.test.ts:496]: Now that `getActive` can return `null`, TS was raising an error.

File: packages/sdk/test/evm/signature-drop-v5.test.ts [hunk 3]
```diff
@@ -515,6 +515,47 @@ describe("Signature drop tests (v5)", async () => {
...
+      await signatureDropContract.createBatch([
+        { name: "test1", description: "test" },
+        { name: "test2", description: "test" },
+      ]);
+      await signatureDropContract.claimConditions.set([
+        {
+          maxClaimableSupply: 1,
+        },
+      ]);
+      // await sdk.updateSignerOrProvider(samWallet);
```
Comment (alexmylonas) [inline packages/sdk/test/evm/signature-drop-v5.test.ts:551]:
```suggestion
```

**Comment by jnsdls** [comment]
@nachoiacovino can you do a manual integration test with this across react-sdk / embeds / dashboard, see if it breaks anything downstream?

**Comment by nachoiacovino** [comment]
hey @alexmylonas seems like the build is failing, can you take a look?

File: packages/sdk/src/evm/core/classes/drop-claim-conditions.ts [hunk 4]
```diff
@@ -747,10 +766,11 @@ export class DropClaimConditions<
     const addressToClaim = address
       ? address
       : await this.contractWrapper.getSignerAddress();
+    const activeCondition = (await this.getActive({ returnInactive: true })) as ClaimCondition;
```
Comment (alexmylonas) [inline packages/sdk/src/evm/core/classes/drop-claim-conditions.ts:769]: Should `returnInactive` here be `true`? If not, how we should fail this request, if there isn't an active claim condition?

File: packages/sdk/test/evm/signature-drop-v4.test.ts (hunk 1 above)
Comment (alexmylonas) [reply packages/sdk/test/evm/signature-drop-v4.test.ts:42]: Good call, updated!

File: packages/sdk/test/evm/signature-drop-v4.test.ts [hunk 5]
```diff
@@ -585,7 +585,7 @@ describe("Signature drop tests (v4)", async () => {
     });
   });
 
-  describe("Claim Conditions", () => {
+  describe("Claim Conditions", () => {
```
Comment (alexmylonas) [inline packages/sdk/test/evm/signature-drop-v4.test.ts:588]:
```suggestion
  describe("Claim Conditions", () => {
```

**Comment by alexmylonas** [comment]
> hey @alexmylonas seems like the build is failing, can you take a look?

@nachoiacovino  It should be good now!

I added a new  field `returnInactive` for [ClaimConditionFetchOptions](https://github.com/alexmylonas/third-web-js/blob/c61788f9cf2408a6bafcb283429818f6715a52ff/packages/sdk/src/evm/types/claim-conditions/claim-conditions.ts#L89) for certain use cases, that might require this.

**Pull request closed without merging**